docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+- Alternative 2: Take in all the predicates given by the user and find using the predicates as a keyword
+
+|     |     |
+|-----|-----|
+|**Pros** | 1. More accurate searching of the ingredient is available for the user.|  
+|**Cons** | 1. Requires users to enter more precise predicate keywords which could be more inconvenient.|
+
+### 4.2. Recipe-related Features
+#### 4.2.1. Addition of recipe
+Users can add a new recipe to the application where there must be at least one or more `ingredient`s. The failure to do so will trigger an exception where the user will be notified of an invalid command and the syntax of the addition of recipe will be displayed. 
+
+> It is important that the name of the new recipe has not appeared in the list of recipes in the application.
+
+When the user attempts to create a new recipe, the `AddRecipeCommand`, ‘Parser’ and `Recipe` class will be accessed and the following sequence of actions are called to create a `recipe` object:
+
+##### Implementation 
+When the user attempts to create a new recipe, the `AddRecipeCommand`, ‘Parser’ and `Recipe` class will be accessed and the following sequence of actions are called to create a `recipe` object:
+
+1. User executes `addrecipe /n Chicken Salad /i Chicken Breast:2:meat, Lettuce:4:vegetable` 
+    1. A `Ui` object will be created and calls `Ui#getUserCommand()`
+    1. Input will be parsed in `Command#parseUserCommand()` and identified with the keyword `addrecipe`.
+    
+    ![Add Recipe Step 1](images/AddRecipe1.png)
+2. Parsing of user input and creation of command object
+    1. This will automatically trigger the parsing of the user’s input string into a suitable format for the addition of `recipe` object in `Command#prepareAddRecipe()`.
+    1. A `AddRecipeCommand` object will be created and calls `AddRecipeCommand#setAttributesOfCmd()` to set the contents of the command into reader friendly formats.
+    
+    ![Add Recipe Step 2](images/AddRecipe2.png)
+3. Executing Command
+    1. The newly created object will call `#AddRecipeCommand#execute` which starts the process of adding a recipe, thus calling `Recipe#AddRecipe()`.
+    1. A `Recipe` object will be created with its name that was parsed in step 2.
+    1. An additional step is included where a check for an existing recipe with the same name is conducted with `#AddRecipeCommand#checkIfRecipeExist()`. A `KitchenHelperException` exception will be triggered when there is an existing recipe.
+    
+    ![Add Recipe Step 3](images/AddRecipe3.png)
+4. `Ingredient`s parsed in step 2 will be added to the newly created recipe according to their category through the calling of `Recipe#addIngredientsToRecipe()`.
+	
+	![Add Recipe Step 4](images/AddRecipe4.png)
+
+All description and warnings to the user utilises the `UI` class, which controls the printing of the text on the console. 
+
+The following sequence diagram shows how the `addrecipe` command works
+
+![AddRecipeCommand](images/addRecipeSequenceDiagram.png) 
```

Comment:
- Operation invoking arrow for setAttributesOfCmd() is slightly misplaced
- Missing return control / return value for setAttributesOfCmd() and :AddRecipeCommand